### PR TITLE
Bugfix: Lightning-Kokkos device kernels cannot use IMAG, ONE, ZERO

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
#640 refactored the Kokkos kernels and used the util functions `IMAG`, `ONE` and `ZERO`. These cannot be used because they are host functions which won't compile for a device execution space.

**Description of the Change:**
Revert to literals and `Kokkos::complex` constant definitions.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-64005] [sc-63455]

### CODE REVIEW

**Comment by AmintorDusko** [comment]
Hey @vincentmr, could you please import [these](https://github.com/PennyLaneAI/pennylane-lightning/pull/741/files#diff-58d52d49250aedf38368981156fb0e026d412877ae0ab93272815595ff8ec202L226-R228) changes?
It is better if we don't pipe out the error signal.

**Review by AmintorDusko — APPROVED** [review]
Hey @vincentmr, thank you for that!

**Review by maliasadi — APPROVED** [review]
Thanks @vincentmr :parachute:

File: .github/workflows/tests_lqcpu_python.yml [hunk 1]
```diff
@@ -11,14 +11,6 @@ on:
         required: true
         description: The version of PennyLane to use. Valid values are either 'release' (most recent release candidate), 'stable' (most recent git-tag) or 'latest' (most recent commit from master)
   pull_request:
-    paths-ignore:
```
Comment (maliasadi) [inline .github/workflows/tests_lqcpu_python.yml:14]: Looks good! :+1:

**Pull request merged**